### PR TITLE
Further document publish command

### LINKED ISSUE #1280: Publish documentation issues
Per @prolativ 

4) There was a misleading comment in the docs about whether to use `central` or `central-s01` as the repository but now I can't find the same guide page I followed while doing the publish
5) there was a mismatch in naming between the documentation and the actually implemented cli options: `publish.vcs` vs `publish.versionControl`

### PR DESCRIPTION
**Pull request by MaciejG604**
Additionaly:
- fix choosing signing mechanism so that `--signer none` always takes priority

### CODE REVIEW

File: website/docs/commands/publishing/publish.md [hunk 1]
```diff
@@ -117,6 +117,9 @@ handled by either
 - the [Bouncy Castle library](https://www.bouncycastle.org) (default, recommended)
 - the local `gpg` binary on your machine
 
+A signing mechanism will be chosen based on options and directives specified,
+it can also be overriden with `--signer` with one of those values: `bc`, `gpg` or `none`.
```
Comment (lwronski) [inline website/docs/commands/publishing/publish.md]: Can you explain the meanings of the values `bc` and `gpg`? Also, what happens when the none option is selected?

File: website/docs/commands/publishing/publish.md [hunk 2]
```diff
@@ -128,25 +131,37 @@ pass it like `--secret-key-password env:MY_KEY_PASSWORD`.
...
+- with options:
+    ```sh
+    scala-cli publish \
+      --secret-key env:PGP_SECRET \
+      --secret-key-password file:pgp_password.txt \
+      …
+    ```
+
+Since these values should be kept secret the options and directives accept the format documented [here](/docs/reference/password-options.md).
+Furthermore command line options accept an extra format - `config:…` for using config entries.
```
Comment (lwronski) [inline website/docs/commands/publishing/publish.md]: Since the default behavior is to retrieve values from the config, do we need to include the `config:..` syntax in this sentence?

File: modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala [hunk 3]
```diff
@@ -248,8 +249,13 @@ object Publish extends ScalaCommand[PublishOptions] with BuildCommandHelpers {
       options.mainClass,
       dummy = options.sharedPublish.dummy
     )
+
+
```
Comment (lwronski) [inline modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala]: Remove extra whitespace lines.

File: modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala [hunk 4]
```diff
@@ -1085,4 +860,214 @@ object Publish extends ScalaCommand[PublishOptions] with BuildCommandHelpers {
         }
     }
   }
+
+  private def buildFileSet(
```
Comment (lwronski) [inline modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala]: Did you make any changes in `buildFileSet` function? I see, that you are new author of this function.
Can you revert this changes to not change git history?

File: website/docs/commands/publishing/publish.md [hunk 5]
```diff
@@ -128,25 +131,37 @@ pass it like `--secret-key-password env:MY_KEY_PASSWORD`.
...
-You can then ask publish to use the key kept in the Scala CLI config with
-```sh
-scala-cli publish \
-  --secret-key config:pgp.secret-key \
-  --secret-key-password config:pgp.secret-key-password \
-  …
-```
+This generated key kept in config will be used by default unless specified otherwise, e.g.:
+- with directives:
+    ```bash
```
Comment (Gedochao) [inline website/docs/commands/publishing/publish.md]:
````suggestion
    ```scala
````

File: website/docs/commands/publishing/publish.md [hunk 6]
```diff
@@ -128,25 +131,37 @@ pass it like `--secret-key-password env:MY_KEY_PASSWORD`.
...
+    //> using publish.secretKey env:PGP_SECRET
+    //> using publish.secretKeyPassword command:get_my_password
+    ```
+
+- with options:
+    ```sh
+    scala-cli publish \
```
Comment (Gedochao) [inline website/docs/commands/publishing/publish.md]:
````suggestion
    ```bash ignore
    scala-cli --power publish \
````

File: website/docs/commands/publishing/publish-setup.md [hunk 7]
```diff
@@ -78,7 +78,7 @@ in Scala CLI (via the `publish.credentials` config key in both cases).
 
 These can be written in the Scala CLI configuration the following way:
 ```sh
```
Comment (Gedochao) [inline website/docs/commands/publishing/publish-setup.md]:
````suggestion
```bash ignore
````

File: website/docs/commands/publishing/publish.md [hunk 8]
```diff
@@ -76,14 +76,14 @@ To override this default value, set the `publish.computeVersion` directive, like
...
 
 A repository is required for the `publish` command, and might need other settings to work fine
-(to pass credentials for example).
+(to pass credentials for example). See [Repositories](#repositories) for more information.
 
 When publishing from you CI, we recommend letting `scala-cli publish setup`
 setting those settings via using directives. When publishing from your local machine to Maven Central,
 we recommend setting the repository via a `publish.repository` directive, and keeping your
 Sonatype credentials in the Scala CLI settings, via commands such as
 ```bash
```
Comment (Gedochao) [inline website/docs/commands/publishing/publish.md]:
````suggestion
```bash ignore
````

File: website/docs/commands/publishing/publish.md [hunk 9]
```diff
@@ -128,25 +131,37 @@ pass it like `--secret-key-password env:MY_KEY_PASSWORD`.
 
 Scala CLI can generate and keep a secret key for you. For that, create the key with
 ```sh
```
Comment (Gedochao) [inline website/docs/commands/publishing/publish.md]:
````suggestion
```bash ignore
````

File: website/docs/commands/publishing/publish.md [hunk 10]
```diff
@@ -128,25 +131,37 @@ pass it like `--secret-key-password env:MY_KEY_PASSWORD`.
...
+
+- with options:
+    ```sh
+    scala-cli publish \
+      --secret-key env:PGP_SECRET \
+      --secret-key-password file:pgp_password.txt \
+      …
+    ```
+
+Since these values should be kept secret the options and directives accept the format documented [here](/docs/reference/password-options.md).
```
Comment (Gedochao) [inline website/docs/commands/publishing/publish.md]:
```suggestion
Since these values should be kept secret, the options and directives accept the format documented [here](/docs/reference/password-options.md).
```

File: website/docs/commands/publishing/publish.md (hunk 2 above)
Comment (Gedochao) [inline website/docs/commands/publishing/publish.md]:
```suggestion
Furthermore, command line options accept an extra format: `config:…` for using config entries.
```

File: website/docs/commands/publishing/publish.md [hunk 11]
```diff
@@ -128,25 +131,37 @@ pass it like `--secret-key-password env:MY_KEY_PASSWORD`.
...
 #### GPG
 
 Using GPG to sign artifacts requires the `gpg` binary to be installed on your system.
 A benefit of using `gpg` to sign artifacts over Bouncy Castle is: you can use keys from
 your GPG key ring, or from external devices that GPG may support.
 
-To enable signing with GPG, pass `--gpg-key *key_id*` on the command line. If needed,
-you can specify arguments meant to be passed to `gpg`, with `--gpg-option`, like
+To enable signing with GPG, pass `--gpg-key *key_id*` on the command line
+or specify it by `//>using publish.gpgKey "key_id"`.
```
Comment (Gedochao) [inline website/docs/commands/publishing/publish.md]:
```suggestion
or specify it with a `using` directive: `//>using publish.gpgKey "key_id"`.
```

File: website/docs/commands/publishing/publish.md [hunk 12]
```diff
@@ -192,9 +207,13 @@ used if it's there. Else the main directive is used.
 
 ### Maven Central
 
+The easiest way right now to publish to Maven Central Repository is to use
```
Comment (Gedochao) [inline website/docs/commands/publishing/publish.md]: NIT
```suggestion
Right now the easiest way to publish to Maven Central Repository is to use
```

File: website/docs/commands/publishing/publish.md [hunk 13]
```diff
@@ -239,13 +258,14 @@ and the [CI overrides](#ci-overrides).
 ## Publishing
 
 Once all the necessary settings are set, publish a Scala CLI project with a command
-such as this one (`.` is for the Scala CLI project in the current directory):
+such as this one:
 
 <ChainedSnippets>
 
 ```sh
 scala-cli publish .
```
Comment (Gedochao) [inline website/docs/commands/publishing/publish.md]:
````suggestion
```bash ignore
scala-cli --power publish .
````

**Review by lwronski — APPROVED** [review]
LGTM
